titus-common/src/main/java/com/netflix/titus/common/util/Evaluators.java

```diff
@@ -118,4 +122,39 @@ public static void times(int count, Consumer<Integer> task) {
         }
         return Optional.empty();
     }
+
+    /**
+     * Remember last function execution result, and return the cached value on subsequent invocations, until the
+     * argument changes.
+     * <h1>
+     * For memoizing suppliers use {@link com.google.common.base.Suppliers#memoize(com.google.common.base.Supplier)}.
+     * </h1>
+     */
+    public static <T, R> Function<T, R> memoizeLast(Function<T, R> function) {
```

Comment:
this seems very similar to `com.netflix.titus.common.util.cache.MemoizedFunction`, can we combine them?

Reply:
I missed that one. Yes, it make sense to combine them. I will extend it to deal with failures as well, so we have two variants available (with error and without error caching).